qcfractal/storage_sockets/sqlalchemy_socket.py

```diff
@@ -1138,9 +1138,12 @@ def add_results(self, record_list: List[ResultRecord]):
 
         meta = add_metadata_template()
 
-        result_ids = []
+        result_ids = ["placeholder"] * len(record_list)
+
+        results_list = []
+        results_idx = []
         with self.session_scope() as session:
-            for result in record_list:
+            for i, result in enumerate(record_list):
 
                 doc = session.query(ResultORM).filter_by(
```

Comment:
This doesn't work for duplicates added in the query since it isn't in the database yet.

Reply:
Changed the code to address the results that were added before.